### PR TITLE
add seo permission

### PR DESCRIPTION
**Pull request by haroun**
Allow non `admin` users to access the SEO tools without granting full access to sensitive admin only features

### CODE REVIEW

**Comment by haroun** [comment]
@ValJed @boutell is this something you think we can merge into the module?

File: lib/api.js [hunk 1]
```diff
@@ -186,4 +186,12 @@ module.exports = (self, { scanRules }) => {
 
     return deadLinks;
   };
+
+  self.addPermissions = function() {
+console.log('>>>>> adding SEO permission')
```
Comment (boutell) [inline lib/api.js]: Please remove log statements.

**Review by boutell — COMMENTED** [review]
@abea what do you think? This is an interesting contribution. If we merge it, it should probably be clarified that it only grants the ability to use the SEO scan button, not the ability to, say, edit just the SEO fields, which people might be looking for. (There are other avenues for that in A2.)

**Comment by haroun** [comment]
@boutell I've implemented the requested changes

File: lib/routes.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ const axios = require('axios');
 module.exports = function (self, options) {
   self.addRoutes = function() {
     self.route('post', 'page-scan-modal', async (req, res) => {
-      if (!self.apos.permissions.can(req, 'admin')) {
+      if (!(self.apos.permissions.can(req, 'admin') || self.apos.permissions.can(req, 'admin-seo'))) {
```
Comment (boutell) [inline lib/routes.js]: A user with the admin permission automatically has all other permissions, so you can check just for the admin-seo permission.

**Review by boutell — CHANGES_REQUESTED** [review]

File: lib/routes.js (hunk 2 above)
Comment (haroun) [reply lib/routes.js]: Done

**Review by boutell — APPROVED** [review]

File: lib/routes.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ const axios = require('axios');
 module.exports = function (self, options) {
   self.addRoutes = function() {
     self.route('post', 'page-scan-modal', async (req, res) => {
-      if (!self.apos.permissions.can(req, 'admin')) {
+      if (!self.apos.permissions.can(req, 'admin-seo')) {
```
Comment (boutell) [inline lib/routes.js]: Let's rename this to "admin-seo-scan" to complete making it clear what it lets you do.

**Review by boutell — CHANGES_REQUESTED** [review]
Also the "label" here should say "SEO scan" so it's clear this doesn't necessarily let you edit SEO fields. Then I think we are good!

File: lib/routes.js (hunk 3 above)
Comment (haroun) [reply lib/routes.js]: I've renamed it "admin-seo-page-scan" to match the name of the feature.
Please let me know if it's ok.

**Comment by boutell** [comment]
Yes, thank you. I'm going to change "admin-seo-page-scan" to just "seo-page-scan" but I won't torture you further by asking you to do it. 😂

**Pull request merged**